fearless/Common/DataProvider/Adapter/AccountInfoSubscriptionAdapter.swift

```diff
@@ -0,0 +1,113 @@
+    func subscribe(chain: ChainModel, accountId: AccountId, handler: AccountInfoSubscriptionAdapterHandler?)
+    func subscribe(chains: [ChainModel], handler: AccountInfoSubscriptionAdapterHandler?)
+
+    func reset()
+}
+
+class AccountInfoSubscriptionAdapter: AccountInfoSubscriptionAdapterProtocol {
+    private weak var handler: AccountInfoSubscriptionAdapterHandler?
+    internal var walletLocalSubscriptionFactory: WalletLocalSubscriptionFactoryProtocol
+    private var accountInfoProviders: [AnyDataProvider<DecodedAccountInfo>]?
+    private var ormlAccountInfoProviders: [AnyDataProvider<DecodedOrmlAccountInfo>]?
+    private var selectedMetaAccount: MetaAccountModel
+
+    init(
+        walletLocalSubscriptionFactory: WalletLocalSubscriptionFactoryProtocol,
+        selectedMetaAccount: MetaAccountModel
+    ) {
+        self.walletLocalSubscriptionFactory = walletLocalSubscriptionFactory
+        self.selectedMetaAccount = selectedMetaAccount
+    }
+
+    func reset() {
+        if let providers = ormlAccountInfoProviders {
+            for provider in providers {
+                provider.removeObserver(self)
+            }
+        }
+
+        if let providers = accountInfoProviders {
+            for provider in providers {
+                provider.removeObserver(self)
+            }
+        }
+    }
+
+    func subscribe(chain: ChainModel, accountId: AccountId, handler: AccountInfoSubscriptionAdapterHandler?) {
+        reset()
+
+        self.handler = handler
+
+        var ormlProviders: [AnyDataProvider<DecodedOrmlAccountInfo>] = []
+        var defaultProviders: [AnyDataProvider<DecodedAccountInfo>] = []
+
+        if chain.chainId.isOrml {
+            if let provider = subscribeToOrmlAccountInfoProvider(for: accountId, chain: chain) {
+                ormlProviders.append(provider)
+            }
+        }
+
+        if !chain.chainId.isOrml {
+            if let provider = subscribeToAccountInfoProvider(for: accountId, chainId: chain.chainId) {
+                defaultProviders.append(provider)
+            }
+        }
+
+        ormlAccountInfoProviders = ormlProviders
+        accountInfoProviders = defaultProviders
+    }
+
+    func subscribe(chains: [ChainModel], handler: AccountInfoSubscriptionAdapterHandler?) {
+        reset()
+
+        self.handler = handler
+
+        var ormlProviders: [AnyDataProvider<DecodedOrmlAccountInfo>] = []
+        var defaultProviders: [AnyDataProvider<DecodedAccountInfo>] = []
+
+        chains.forEach { chain in
+            if chain.chainId.isOrml, let accountId = selectedMetaAccount.fetch(for: chain.accountRequest())?.accountId {
+                if let provider = subscribeToOrmlAccountInfoProvider(for: accountId, chain: chain) {
+                    ormlProviders.append(provider)
+                }
+            }
+
+            if !chain.chainId.isOrml, let accountId = selectedMetaAccount.fetch(for: chain.accountRequest())?.accountId {
```

Comment:
Again, `else` instead. And you can definitely unwrap `accountId` before `if/else`